modal/app.py

```diff
@@ -374,16 +373,6 @@ async def run(
         """
         from .runner import _run_app  # Defer import of runner.py, which imports a lot from Rich
 
-        # See Github discussion here: https://github.com/modal-labs/modal-client/pull/2030#issuecomment-2237266186
-
-        if show_progress is True:
-            deprecation_error(
-                (2024, 11, 20),
-                "`show_progress=True` is no longer supported. Use `with modal.enable_output():` instead.",
-            )
-        elif show_progress is False:
-            deprecation_warning((2024, 11, 20), "`show_progress=False` is deprecated (and has no effect)")
```

Comment:
Just removing this entirely instead of making it a `deprecation_error` since it had no effect anyway.